.github/workflows/pr-run-checks.yml

```diff
@@ -0,0 +1,20 @@
+name: pr-run-checks
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  check-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 15.x, 16.x]
```

Comment:
This is awesome. `strategy.matrix` basically runs the build multiple times with each variable.

For example, we could potentially do something like
```yml
strategy:
  matrix:
      node-version: [12.x, 13.x]
      eslint-version: [6.0, 7.0]
```

This will run the build 4 times for each combination of variables